File: Python/Bubble sort.py
"""
The logic is to swap one element with its next element if they are not in order.
We do this n times for all elements and array is sorted.
Note:We get the Ith maximum number after each iteration in the right side of array
We can improve this by going till the unsorted elements i.e-(n-1-i)
We can further improve by stopping if there have been no swaps in one iteration
"""
def bubbleSort(lst,n):
    for i in range(n-1):
        flag=True
        for j in range(n-1-i):
            if lst[j]>lst[j+1]:
                lst[j],lst[j+1]=lst[j+1],lst[j]
                flag=False
        if flag:
            break
    return lst

lst=[1,5,2,19,12,11,10]
print(bubbleSort(lst,len(lst)))

File: Java/BubbleSort.java
//The Logic is to swap every element with its next element if they are not in correct order.
//If we do this n times we get the sorted array
//Note:We get ith largest element at correct position after ith iteration
//We can improve this logic by going till the unsorted part of array i.e: n-1-i
//We can further improve by stopping the process if there have been no swaps in one iteration

class BubbleSort
{
	public static void main(String args[])
	{
		int arr[]={10,1,5,18,11,12};
		int n=arr.length;
		for(int i=0;i<n;i++)
		{
			Boolean flag=true;
			for(int j=0;j<n-1-i;j++)
			{
				if(arr[j]>arr[j+1])
				{
					flag=false;
					int temp=arr[j];
					arr[j]=arr[j+1];
					arr[j+1]=temp;
				}
			}
			if(flag)
				break;
			//Print.array(arr,n);
		}
		Print.array(arr,n);
	}	
}
class Print
{
	static void array(int arr[],int n)
	{
		for(int i=0;i<n;i++)
		{
			System.out.print(arr[i]);
			System.out.print(" ");
		}
		System.out.println();

	}
}